gm4_smelteries/data/gm4_smelteries/functions/find_furnace.mcfunction

```diff
@@ -8,4 +8,6 @@ execute if block ~-1 ~ ~ furnace[facing=west] run tp @s ~ ~ ~ 90 0
 execute if block ~ ~ ~1 furnace[facing=south] run tp @s ~ ~ ~ 0 0
 execute if block ~ ~ ~-1 furnace[facing=north] run tp @s ~ ~ ~ 180 0
 execute at @s if block ^ ^ ^1 furnace run tag @s add gm4_bf_has_furnace
-execute at @s unless block ^ ^ ^1 furnace run tp @s ~ ~ ~ 45 0
+execute at @s if block ^ ^ ^1 furnace run data merge entity @s {HandItems:[{},{id:"minecraft:stone_button",Count:1b,tag:{CustomModelData:4}}],Pose:{Head:[0f, 0f, 0f],LeftArm:[0f, 0f, 0f]}}
+execute at @s unless block ^ ^ ^1 furnace run tp @s ~ ~ ~ -45 0
+execute at @s unless block ^ ^ ^1 furnace run data merge entity @s {HandItems:[{},{id:"minecraft:stone_button",Count:1b,tag:{CustomModelData:5}}],Pose:{Head:[0f, 0f, 0f],LeftArm:[0f, 0f, 0f]}}
```

Comment:
This needs to be implemented more carefully; `find_furnace` is clocked at 1.25Hz so we need to limit costly operation such as NBT operations.
This item updating only needs to happen once, right? As in, once the armor stand is holding the correct button it doesn't need be given the same button again 16 ticks later.
We should limit these NBT operations so that they only happen once the "gm4_bf_has_furnace" state changes.

Reply:
I *think* I took care of it... at least it still works.